internal/shellgen/flake_package.go

```diff
@@ -0,0 +1,66 @@
+package shellgen
+
+import (
+	"path/filepath"
+	"strings"
+
+	"go.jetpack.io/devbox/internal/boxcli/featureflag"
+	"go.jetpack.io/devbox/internal/nix"
+)
+
+type flakePackage struct {
```

Comment:
I'm not sure if `flakePackage` is an accurate name. It's fetching a store path and its dependencies, which may or may not have been built from a flake.

Reply:
I'm a bit confused, why do we need this instead of using `nix.Package` directly? It feels like just a another struct with the same data and function?